client.go

```diff
@@ -58,6 +58,23 @@ func NewClient(url string, http Doer) (Client, error) {
 	}, nil
 }
 
+// NewRequest builds a data sink compatable http.Request.
+func NewRequest(requestURL string, stream Stream, msg Message) (*http.Request, error) {
+	path := requestURL + endpoint + "/" + url.PathEscape(stream.ID)
+	if stream.PartitionKey != "" {
+		path += "?partition_key=" + url.QueryEscape(stream.PartitionKey)
+	}
+
+	req, err := http.NewRequest(http.MethodPost, path, bytes.NewBuffer(msg))
+
+	if err != nil {
```

Comment:
Minor: the newline before this makes it look a bit odd. It's preferred to have the `if err != nil` check immediately follow the line that populates `err`.